### PR TITLE
Updated Order related components

### PR DESCRIPTION
**Pull request by petro97**
Description:

Linting:

- [ ] No linting errors

Tests:

- [ ] E2E tests (npm test run with `e2e`)
- [ ] Manual tests

Documentation:

- [ ] Requires documentation updates

### CODE REVIEW

File: components/src/CartLineItem/cart.lineitem.tsx [hunk 1]
```diff
@@ -93,33 +102,20 @@ class CartLineItem extends React.Component<CartLineItemProps, CartLineItemState>
...
   }
 
-  handleQuantityChange(event) {
+  async handleQuantityChange(event) {
     event.preventDefault();
     const { item, handleQuantityChange } = this.props;
     const { quantity } = this.state;
     if (quantity === '') {
-      this.setState({ quantity: 1 });
+      this.setState({ quantity: '1' });
```
Comment (dusanradovanovic) [inline components/src/CartLineItem/cart.lineitem.tsx]: Why is quantity a string here?

File: components/src/CartLineItem/cart.lineitem.tsx [hunk 2]
```diff
@@ -136,90 +132,49 @@ class CartLineItem extends React.Component<CartLineItemProps, CartLineItemState>
...
-              handleErrorMessage(json);
-            });
-          }
-        })
-        .catch((error) => {
-          // eslint-disable-next-line no-console
-          console.error(error.message);
-        });
-    });
+    const quantity = itemQuantity || 1;
```
Comment (dusanradovanovic) [inline components/src/CartLineItem/cart.lineitem.tsx]: Quantity is a number here. It should be consistent and be either a number all the time or a string all the time.

File: components/src/CategoryItems/categoryitems.main.tsx [hunk 3]
```diff
@@ -118,62 +204,51 @@ class CategoryItemsMain extends React.Component<CategoryItemsMainProps, Category
...
+    const dataCode = {
+      code: categoryId,
+    };
+    try {
+      const navigationLookupFormRes = await this.client.root().fetch({ lookups: { navigationlookupform: {} } });
+      const categoryDataRes = await navigationLookupFormRes.lookups.navigationlookupform(dataCode).fetch(zoomNavigation);
+      this.setState({
+        categoryModel: categoryDataRes,
+        categoryModelDisplayName: categoryDataRes.displayName,
+        // categoryModelParentDisplayName: categoryDataRes.parent ? categoryDataRes.parent[0].displayName : '',
```
Comment (dusanradovanovic) [inline components/src/CategoryItems/categoryitems.main.tsx]: Dead code

File: components/src/ProductListPagination/productlistpagination.main.tsx [hunk 4]
```diff
@@ -71,17 +71,17 @@ class ProductListPagination extends React.Component<ProductListPaginationProps,
...
-    }
-    for (let i = 0; i < paginationData.links.length; i++) {
-      if (paginationData.links[i].rel === 'previous') {
-        paginationPreviousLinkVar = paginationData.links[i].uri;
-      }
-      if (paginationData.links[i].rel === 'next') {
-        paginationNextLinkVar = paginationData.links[i].uri;
-      }
-    }
+    // if (paginationData.self.type.includes('navigation')) {
```
Comment (dusanradovanovic) [inline components/src/ProductListPagination/productlistpagination.main.tsx]: Dead code

File: app/src/containers/CartPage.tsx [hunk 5]
```diff
@@ -222,12 +238,12 @@ class CartPage extends React.Component<RouteComponentProps, CartPageState> {
               <div>
                 <div className="cart-sidebar-inner">
                   <div data-region="cartSummaryRegion" className="cart-summary-container" style={{ display: 'inline-block' }}>
-                    <AddPromotionContainer data={cartData} onSubmittedPromotion={() => { this.fetchCartData(); }} />
+                    <AddPromotionContainer onSubmittedPromotion={() => { this.fetchCartData(); }} />
```
[the remark this answers is not in the document]
Comment (rostyk-kanafotskyy) [reply app/src/containers/CartPage.tsx]: Thanks for noticing 🙂

File: app/src/containers/PurchaseReceiptPage.tsx [hunk 6]
```diff
@@ -42,7 +42,7 @@ const PurchaseReceiptPage = (props: RouteComponentProps) => {
           <h2>
             {intl.get('order-processed-message')}
           </h2>
-          <PurchaseDetailsMain data={location.state.data} onReorderAllProducts={handleReorderAllProducts} itemDetailLink={itemDetailLink} />
+          <PurchaseDetailsMain data={JSON.parse(location.state.data)} onReorderAllProducts={handleReorderAllProducts} itemDetailLink={itemDetailLink} />
```
[the remark this answers is not in the document]
Comment (rostyk-kanafotskyy) [reply app/src/containers/PurchaseReceiptPage.tsx]: `location.state.data` passed from `OrderReviewPage` contains a function as a field `addtocartform`. It brakes a `history.push` 🤔 

I updated zoom not to send functions as a response, so now it works without `JSON.parce`.
Thanks for bringing that up.

File: components/src/CartLineItem/cart.lineitem.tsx (hunk 1 above)
Comment (rostyk-kanafotskyy) [reply components/src/CartLineItem/cart.lineitem.tsx]: it's a mistake.
Updated it.

**Pull request merged**